src/plugins/intel_npu/src/plugin/npuw/just_sync_infer_request.cpp

```diff
@@ -453,7 +488,7 @@ void ov::npuw::JustInferRequest::unpack_closure(std::size_t idx, RqPtr request)
         const auto closure_param_id = comp_model_desc.param_base + cidx;
         auto& iport = func_desc.compiled_model->inputs()[closure_param_id];
         auto clparam = request->get_tensor(iport);
-        if (closure.get_element_type() != clparam->get_element_type()) {
+        if (comp_model_desc.unpack_required[cidx] && closure.get_element_type() != clparam->get_element_type()) {
```

Comment:
so I believe the 2nd check can be omitted?

Reply:
..and the underlying `else` should be extended as `else if (update_required)` and then do set_tensor

Reply:
Done